ores/scoring_systems/scoring_system.py

```diff
@@ -17,12 +18,16 @@
 class ScoringSystem(dict):
 
     def __init__(self, context_map, score_cache=None,
-                       metrics_collector=None, timeout=None):
+                       metrics_collector=None, timeout=None, lock_manager=None,
+                       connections_per_ip=4, connections_per_ip_hard=7):
```

Comment:
Feels like we these last parameters belong to the lock_manager and shouldn't be exposed to ScoringSystem.

Reply:
no, lock manager can be used for anything, for example lock manager can be used to lock a certain score so other workers don't score it. In general it's a little bit YAGNI, but it shows these are conceptually different. You can see that Thumbor also followed this line of thought which I agree with.